frontend/src/components/CodeContainer/index.tsx

```diff
@@ -34,17 +34,38 @@ type CodeContainerProps = {
 export function CodeContainer(props: CodeContainerProps) {
   const { context, line_from, sub_matches, line_to } = props;
   const [codeLineFrom, setCodeLineFrom] = useMountedState(line_from);
-  const [codeLineTo, setCodeLineTo] = useMountedState(line_to);
+  const [codeLineTo, setCodeLineTo] = useMountedState(0);
   const [code, setCode] = useMountedState(props.context.snippet);
 
   const loadUpperCode = () => {
-    setCodeLineFrom(1);
-    setCode(`${context.upper_lines}${code}`);
+    const upperCodeArray = context.upper_lines.split("\n");
+    const upperCode = upperCodeArray
+      .slice(codeLineFrom - 11 > 0 ? codeLineFrom - 11 : 0, codeLineFrom)
```

Comment:
ideally, I would replace 10 with some constant and 11 use as `constant+1`. So in future you could change the number of loading lines just by changing number in one place and avoid "magic number" anti-pattern